shlomobot_pytest/common_tests.py

```diff
@@ -79,63 +84,57 @@ def find_functions_with_single_quote_docstrings(file_list: list[str]) -> list[st
     return single_quote_docstrings
 
 
-def contains_main_function(module_name: str) -> bool:
+def contains_main_function(py_filename: str) -> bool:
     """
     Checks if the 'main' function exists within module
     """
-    module = import_pyfile(module_name)
+    module = import_pyfile(py_filename)
     try:
         callable(getattr(module, "main"))
         return True
     except AttributeError:
         return False
 
 
-def builtins_not_used_as_variable(file_list: list[str]) -> bool:
+def builtins_not_used_as_variable(function: FunctionType) -> bool:
     """
-    Checks if inside the files given in 'file_list' any builtins are used as variables
-
-    Return True if no builtins are used as variables, False if at least one is.
+    Return True if no builtins are used as variables in the function, else return True
     """
-    builtins_list = [word for word in dir(builtins) if not re.match("[A-Z|_]", word[0])]
-    functions_list = get_functions_from_files(file_list)
-
     builtin_used_as_variable = r"\n\s*(?:[^\s]*? ?, ?)*?{0} ?(?:\s*,\s*[^\W]+?\s*)*=.*"
     builtin_given_to_function = r"def {0}\((?:[^\s]*? ?, ?)*?{1}(?:\s*,\s*[^\W]+?)*\):"
+    builtins_list = [word for word in dir(builtins) if not re.match("[A-Z|_]", word[0])]
```

Comment:
you don't need re.match since you are only checking on letter

```suggestion
    builtins_list = [word for word in dir(builtins) if word[0] not in "[A-Z|_]"]
```

Reply:
(`import string` for my suggestion)